### PR TITLE
doc(README.md, doc): add design and architecture documents

### PR DESCRIPTION
**Pull request by arschles**
Fixes https://github.com/deis/k8s-claimer/issues/6

### CODE REVIEW

File: doc/architecture.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# k8s-claimer architecture
+
+This document details how the k8s claimer server performs its lease and release operations in
+accordance with the guarantees it gives.
+
+## General
+
+Each cluster is assigned a unique name, and the GKE API/backend stores those names (along
```
Comment (jackfrancis) [inline doc/architecture.md:8]: I think we mean each "lease" is assigned a unique name, right?

Comment (arschles) [inline doc/architecture.md:8]: Nope, each lease is assigned a uuid (referred to as a token in the design doc). The cluster is assigned a static (but unique) name, which doesn't change between leases.

**Comment by helgi** [comment]
Looks good, but would like to see who is leasing it.

Here is how we did clouds and stacks before:

```
Helgi Þorbjörnsson [09:30] 
eybot stack me

eybot BOT [09:30] 
you are assigned earth-2012 (like stable-v4: gentoo-2012) for 12 hours.

Helgi Þorbjörnsson [09:30] 
eybot stacks

eybot BOT [09:30] 
available: jupiter, mars, mercury, neptune, pluto, saturn, ubuntu-dev1, ubuntu-dev2, ubuntu-dev3, ubuntu-dev4, ubuntu-dev5, ubuntu-dev6, venus-2012

Helgi Þorbjörnsson [09:30] 
eybot stacks all

Helgi Þorbjörnsson [09:32] 
eybot stack release

eybot BOT [09:32] 
released assignment for: earth-2012.
```

You could also request a particular stack in that scenario but that's beside the point. Then you can see if someone has multiple `leases` and force release them if required.

**Comment by arschles** [comment]
@helgi I didn't think about assigning identity to a cluster in this initial design. It's just whoever has the cluster uuid can release it. Since this _initial_ design doesn't call for it to have a chat interface (the plan is to have it programmatically called by e2e test runs), I don't believe it's worth it to add identity right now. Thoughts?

**Comment by helgi** [comment]
Probably isn't worth it off the bat, I just want to make it something you account for early on in the arch. Adding a bot layer with Slack these days is so easy anyway.

**Comment by arschles** [comment]
@helgi now that you bring it up, I think it would be fairly easy to add identity to a claim, again using an annotation. Identity could go into a second key/value pair. For example:

``` console
UUID => (cluster_name, lease_expiration_time)
identity_name => UUID
```

The first line is what's in this design, and the second is the addition...

**Comment by arschles** [comment]
@helgi I've created https://github.com/deis/k8s-claimer/issues/11 to track identity (but not slack integration or other chat support)

**Pull request merged**